crates/eigen-cli/src/lib.rs

```diff
@@ -138,40 +261,59 @@ mod test {
-        let output_dir = tempdir().unwrap();
-        let output_path = output_dir.path();
-        let subcommand = EigenKeyCommand::Generate {
-            key_type: key_type.clone(),
-            num_keys: 1,
-            output_dir: output_path.to_str().map(String::from),
+    #[case(BlsKeystoreType::Scrypt)]
+    #[case(BlsKeystoreType::Pbkdf2)]
+    fn test_blskeystore_generation(#[case] keystore_type: BlsKeystoreType) {
+        let subcommand = EigenKeyCommand::BlsConvert {
+            key_type: keystore_type,
+            secret_key: "000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f"
+                .to_string(),
+            output_path: "./keystore.json".to_string(),
+            password: Some("testpassword".to_string()),
         };
+
         let command = Commands::EigenKey { subcommand };
 
         execute_command(command).unwrap();
 
-        let private_key_hex = fs::read_to_string(output_path.join(PRIVATE_KEY_HEX_FILE)).unwrap();
-        let password = fs::read_to_string(output_path.join(PASSWORD_FILE)).unwrap();
-        let key_name = match key_type {
-            KeyType::Ecdsa => "ecdsa",
-            KeyType::Bls => "bls",
-        };
-        let key_path = output_path
-            .join(DEFAULT_KEY_FOLDER)
-            .join(format!("1.{}.key.json", key_name));
+        std::process::Command::new("chmod")
+            .arg("u+w") // Add write permission for the user (owner)
+            .arg("./keystore.json")
+            .status()
+            .expect("Failed to change file permissions");
 
-        let decrypted_bytes = decrypt_key(key_path, password).unwrap();
-        let decrypted_private_key = SecretKey::from_slice(&decrypted_bytes).unwrap().to_bytes();
+        let keystore_instance = Keystore::from_file("./keystore.json").unwrap();
+        let decrypted_key = keystore_instance.decrypt("testpassword").unwrap();
+        let original_secret_key =
+            hex::decode("000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f")
+                .unwrap();
+        assert_eq!(original_secret_key.as_slice(), decrypted_key);
+        std::fs::remove_file("./keystore.json").unwrap();
+    }
 
-        let private_key = hex::decode(private_key_hex).unwrap();
+    #[rstest]
+    #[case(MnemonicLanguage::English)]
+    #[case(MnemonicLanguage::Italian)]
+    #[case(MnemonicLanguage::ChineseSimplified)]
+    #[case(MnemonicLanguage::ChineseTraditional)]
+    #[case(MnemonicLanguage::Spanish)]
+    #[case(MnemonicLanguage::Korean)]
+    #[case(MnemonicLanguage::Portuguese)]
+    #[case(MnemonicLanguage::Czech)]
+    fn test_new_mnemonic(#[case] language: MnemonicLanguage) {
+        let subcommand = EigenKeyCommand::CreateNewMnemonic {
+            language,
+            path: None,
+        };
 
-        assert_eq!(private_key, decrypted_private_key.as_slice());
+        let command = Commands::EigenKey { subcommand };
+
+        execute_command(command).unwrap();
```

Comment:
shouldn't this assert on something?

Reply:
the mnemonic just prints in cli
so nothing to assert 
```
 println!("New mnemonic generated : {}", mnemonic);
            println!("{}", "Please store it safely!".red().bold());
```